modules/site-health/audit-full-page-cache/load.php

```diff
@@ -0,0 +1,198 @@
+
+	switch ( $page_cache_detail['status'] ) {
+		case 'recommended':
+			$default_response['badge']['color'] = 'orange';
+			$default_response['label']          = __( 'Page caching is not detected but the server response time is OK', 'performance-lab' );
+			break;
+		case 'good':
+			$default_response['badge']['color'] = 'green';
+			$default_response['label']          = __( 'Page caching is detected and the server response time is good', 'performance-lab' );
+			break;
+		default:
+			$default_response['badge']['color'] = 'red';
+			$default_response['label']          = __( 'Page caching is detected but the server response time is still slow', 'performance-lab' );
+			if ( empty( $page_cache_detail['headers'] ) && ! $page_cache_detail['advanced_cache_present'] ) {
+				$default_response['label'] = __( 'Page caching is not detected and the server response time is slow', 'performance-lab' );
+			}
+	}
+
+	$page_cache_test_summary = array();
+
+	if ( empty( $page_cache_detail['response_time'] ) ) {
+		$page_cache_test_summary[] = '<span class="dashicons dashicons-dismiss"></span> ' . __( 'Server response time could not be determined. Verify that loopback requests are working.', 'performance-lab' );
+	} else {
+
+		$threshold = perflab_afpc_get_good_response_time_threshold();
+		if ( $page_cache_detail['response_time'] < $threshold ) {
+			$page_cache_test_summary[] = '<span class="dashicons dashicons-yes-alt"></span> ' . sprintf(
+				/* translators: %d is the response time in milliseconds */
+				__( 'Median server response time was %1$s milliseconds. This is less than the %2$s millisecond threshold.', 'performance-lab' ),
+				number_format_i18n( $page_cache_detail['response_time'] ),
+				number_format_i18n( $threshold )
+			);
+		} else {
+			$page_cache_test_summary[] = '<span class="dashicons dashicons-warning"></span> ' . sprintf(
+				/* translators: %d is the response time in milliseconds */
+				__( 'Median server response time was %1$s milliseconds. It should be less than %2$s milliseconds.', 'performance-lab' ),
+				number_format_i18n( $page_cache_detail['response_time'] ),
+				number_format_i18n( $threshold )
+			);
+		}
+
+		if ( empty( $page_cache_detail['headers'] ) ) {
+			$page_cache_test_summary[] = '<span class="dashicons dashicons-warning"></span> ' . __( 'No client caching response headers were detected.', 'performance-lab' );
+		} else {
+			$page_cache_test_summary[] = '<span class="dashicons dashicons-yes-alt"></span> ' .
+										sprintf(
+										/* translators: Placeholder is number of caching headers */
+											_n(
+												'There was %d client caching response header detected:',
+												'There were %d client caching response headers detected:',
+												count( $page_cache_detail['headers'] ),
+												'performance-lab'
+											),
+											count( $page_cache_detail['headers'] )
+										) . ' <code>' . implode( '</code>, <code>', $page_cache_detail['headers'] ) . '</code>.';
```

Comment:
This code looks quite weird. Can we break it down into two assignments to a temporary string variable, and then afterwards assign that string to `$page_cache_test_summary[]`?

Reply:
makes sense